swap/src/bitcoin/wallet.rs

```diff
@@ -36,15 +35,14 @@ impl Wallet {
         electrum_http_url: Url,
         network: bitcoin::Network,
         datadir: &Path,
+        p_key: bitcoin::PrivateKey,
     ) -> Result<Self> {
         // todo: Implement conversion to anyhow::error so we can use ?
         let client =
             Client::new(electrum_rpc_url.as_str()).expect("Failed to init electrum rpc client");
 
         let db = bdk::sled::open(datadir)?.open_tree(SLED_TREE_NAME)?;
 
-        // todo: make key generation configurable using a descriptor
-        let p_key = ::bitcoin::PrivateKey::generate_default()?;
         let bdk_wallet = bdk::Wallet::new(
             bdk::template::P2WPKH(p_key),
```

Comment:
This TODO should be solved by using a descriptor that supports derivation. Most likely BIP84 so you are on native-segwit: https://docs.rs/bdk/0.3.0/bdk/descriptor/template/struct.BIP84.html

Reply:
You will need an `ExtendedPrivKey` as an argument then instead of a normal `PrivateKey`.

Reply:
Sweet, then it is exactly like the Nectar setup :)
We did use Nectar as an inspiration yesterday, but did not fully think it through and only used the private key, not the extended one in the end. I will see to fix it.